[Prototype] NFTs table

katspaugh opened the pull request:
## What it solves
A prototype to display NFTs as a table

## Screenshots
<img width="1182" alt="Screenshot 2023-01-03 at 16 39 28" src="https://user-images.githubusercontent.com/381895/210390341-a64db1df-b3ae-418d-ab5d-56c6a8d170c5.png">

usame-algan commented:
Nice, it looks much more structured than the current view imo! Would be interesting to explore how we could display the images e.g. show them as a "moving" tooltip on hover.

katspaugh commented:
I was thinking that too, but OpenSea is rate-limited at 4 req/sec...

schmanu commented on src/components/nfts/NftGrid/index.tsx:
```diff
@@ -1,57 +1,204 @@
+              </TableRow>
+            ))}
+
+            {/* Fill up the table up to min rows */}
```
Why do we need these min rows?

Is it related to the InfiniteScrolling?

katspaugh replied:
Yes. When you filter, and there are fewer than 10 rows, we don't want the page to jump and trigger InfiniteScroll.

schmanu commented on src/components/nfts/NftCollections/index.tsx:
```diff
@@ -1,61 +1,109 @@
+      {allNfts?.length > 0 && (
+        <>
+          {/* Mass send button */}
+          <Box mb={2} display="flex" alignItems="center" gap={1}>
```
Should we make this sticky?
Especially useful with InfiniteScrolling IMO.

katspaugh replied:
Made it sticky, thanks for the idea!

schmanu commented on src/components/nfts/NftGrid/index.tsx:
```diff
@@ -1,57 +1,204 @@
+
+const minRows = 10
+const iconSize = 20
+const iconStyle = { width: '100%', maxHeight: '100%' }
```
This stretches some images if their original height is bigger than their width.

For instance:
This image
![Original](https://assets.otherside.xyz/otherdeeds/790b73738484af857ab26788630164ed8ed09ce5e582c79a4cc177000a5b5481.jpg)

Looks like this
![Screenshot 2023-01-19 at 16 22 59](https://user-images.githubusercontent.com/2670790/213481892-b6234ea3-21ad-4af6-9fa4-566ba2ef69ef.png)

katspaugh commented:
@schmanu fixed everything, thanks!
Regarding the sticky header, let's do it in another PR and also make the send form a shared component.

schmanu reviewed (approved):
Nice! 🚀

liliya-soroka commented:
1. It's strange to see the message when no opensea app 
<img width="1176" alt="image" src="https://user-images.githubusercontent.com/338622/214071306-89dd30d0-6654-4b2c-a5f4-d7f135f5f303.png">

katspaugh commented:
Fixed. ✅

usame-algan commented:
Is it possible to align these white boxes?
<img width="497" alt="Screenshot 2023-01-23 at 16 48 49" src="https://user-images.githubusercontent.com/5880855/214084102-30b70448-37a5-41e2-8092-dcb2304dd7ec.png">

I noticed that there is no loading state
<img width="1512" alt="Screenshot 2023-01-23 at 16 43 07" src="https://user-images.githubusercontent.com/5880855/214084208-637f28c7-5304-4a52-937e-d68e7e9eb3ef.png">

When not an owner or not connected with a wallet, it is still possible to select/deselect tokens without being able to do any other action. Is this intended or should we hide the filter and select boxes unless an owner is connected?

liliya-soroka commented:
2. https://nfts--webcore.review-web-core.5afe.dev/gor:0xC642bEc0e4f69815e89723C09caA8DAa524cAc9c/balances/nfts - Goerli
children instead of token ID
<img width="532" alt="image" src="https://user-images.githubusercontent.com/338622/214085732-1e1a5668-5bd7-4579-bdf0-1f34941e54f9.png">

katspaugh commented:
Fixed, thanks!

@usame-algan reg. the selection being always enabled, I don't see why not. You can select a few items, and then connect a wallet.

liliya-soroka commented:
GA tracking for new NFT page ?

liliya-soroka commented:
3. I want to leave it here as a comment - the table is not resized if there are no safe-apps for the NFT for the selected chain
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/338622/214248982-4e21fdb6-0008-4a4b-bdf3-b19caae70567.png">

liliya-soroka commented:
5. Send NFT from the side menu is not opened if the user is located on the NFTs page:
- open NFT page
- click New tx in the side menu
- select send NFT
Current result: Nothing happens

liliya-soroka commented:
6. Add the "Batched transactions" title before the list of actions as we have for the Batch option on the history 

<img width="608" alt="image" src="https://user-images.githubusercontent.com/338622/214258900-7e908329-823f-419f-9171-cb6222253973.png">

<img width="593" alt="image" src="https://user-images.githubusercontent.com/338622/214258945-e8bc2870-6690-4819-a56f-40c29f9a3609.png">

katspaugh commented:
* I've added titles in tx modals + a scrollable area for the batched items in the modal
* Added tracking on the Send button (tracks how many NFTs are sent)
* Removed the Send NFT button from the New Transaction modal
* The Send button in the Address Book now opens the Send Token modal directly

liliya-soroka commented:
regression after the fix : no send NFTs when "New transaction" is used on transactions, assets(tokens list) or any tab.
I think it make sense to hide only for the NFTs list or don't hide it at all  
<img width="1317" alt="image" src="https://user-images.githubusercontent.com/338622/214287917-f30fce92-2631-4483-8246-c23cacfbba09.png">

katspaugh commented:
Well, it was intentional, but I like the idea of not showing it only on the NFT page.

katspaugh commented:
Restored the button 👍

liliya-soroka commented:
Could we keep the same view for the long list of actions for NFTs butch , as we have for usual txs with many actions ?
NFTs:
<img width="600" alt="image" src="https://user-images.githubusercontent.com/338622/214841923-b387d11a-dd72-4fac-9bcd-4354d7a720f6.png">

Usual actions list in the modal window 
<img width="590" alt="image" src="https://user-images.githubusercontent.com/338622/214842096-8a6cbcbf-7433-4d9d-ae83-371cb8dc88f3.png">

liliya-soroka commented:
Also question to the table when the list is short:
Tokens list :
<img width="961" alt="image" src="https://user-images.githubusercontent.com/338622/214852912-da5520f7-4aa4-40ef-ba01-71ca3c8e6871.png">

NFTs list:
<img width="876" alt="image" src="https://user-images.githubusercontent.com/338622/214853125-1b59cc78-022a-4cda-aa3d-ca57454b5249.png">

katspaugh commented:
@liliya-soroka I've removed the scroll in the review modal. It was better with it, but I agree it should be consistent with other modals.

Also removed the empty rows in the table. They'll be shown only when filtering to avoid having the page jump.

liliya-soroka commented:
Verified
